### PR TITLE
Add AIE1 startup code

### PR DESCRIPTION
**Pull request by stephenneuendorffer**

### CODE REVIEW

File: libc/startup/baremetal/aie/crt1.cc [hunk 1]
```diff
@@ -0,0 +1,15 @@
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// (c) Copyright 2024 Advanced Micro Devices, Inc. or its affiliates
+
+extern "C" {
+extern int _main(int, char **);
```
Comment (SagarMaheshwari99) [inline libc/startup/baremetal/aie/crt1.cc]: This should be "main"?

Comment (stephenneuendorffer) [reply libc/startup/baremetal/aie/crt1.cc]: Indeed!

File: libc/startup/baremetal/aie/crt1.cc [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// (c) Copyright 2024 Advanced Micro Devices, Inc. or its affiliates
+
+extern "C" {
+extern int main(int, char **);
+void _Exit(int val) {
+  (void)val;
+  done();
+}
```
Comment (SagarMaheshwari99) [inline libc/startup/baremetal/aie/crt1.cc]:
```suggestion
void _Exit(int) { done(); }
```

File: llvm/lib/Target/AIE/AsmParser/AIE1AsmParser.cpp [hunk 3]
```diff
@@ -0,0 +1,512 @@
+//===-- AIEAsmParser.cpp - Assembly Parser for AIE -------- ------*- C++ -*--//
```
Comment (SagarMaheshwari99) [inline llvm/lib/Target/AIE/AsmParser/AIE1AsmParser.cpp]: AIE`1`AsmParser.cpp...

Is this file a copy of the AIE2AsmParser.cpp ? If so. can we keep code common?

Comment (stephenneuendorffer) [reply llvm/lib/Target/AIE/AsmParser/AIE1AsmParser.cpp]: unfortunately there are a number of minor differences which are somewhat difficult to factor out.

File: llvm/lib/Target/AIE/AsmParser/AIEBaseOperand.h [hunk 4]
```diff
@@ -58,6 +58,11 @@ struct AIEBaseOperand : public MCParsedAsmOperand {
   bool isReg() const override { return Kind == Register; };
   bool isMem() const override { llvm_unreachable("No isMem"); };
 
+  /// isCallSymbol - Is this a valid Call Symbol?
+  bool isCallSymbol() const {
```
Comment (konstantinschwarz) [inline llvm/lib/Target/AIE/AsmParser/AIEBaseOperand.h:63]: What is this needed for?

**Review by konstantinschwarz — APPROVED** [review]

File: llvm/lib/Target/AIE/AsmParser/AIEBaseOperand.h (hunk 4 above)
Comment (stephenneuendorffer) [reply llvm/lib/Target/AIE/AsmParser/AIEBaseOperand.h:63]: It's not.  Copypasta!

Comment (stephenneuendorffer) [reply llvm/lib/Target/AIE/AsmParser/AIEBaseOperand.h:63]: scratch that, reverse it.  It's used by the generated matcher:
```
static unsigned validateOperandClass(MCParsedAsmOperand &GOp, MatchClassKind Kind) {
  AIEOperand &Operand = (AIEOperand &)GOp;
  if (Kind == InvalidMatchClass)
    return MCTargetAsmParser::Match_InvalidOperand;

  if (Operand.isToken() && Kind <= MCK_LAST_TOKEN)
    return isSubclass(matchTokenString(Operand.getToken()), Kind) ?
             MCTargetAsmParser::Match_Success :
             MCTargetAsmParser::Match_InvalidOperand;

  switch (Kind) {
  default: break;
  // 'CallSymbol' class
  case MCK_CallSymbol: {
    DiagnosticPredicate DP(Operand.isCallSymbol());
    if (DP.isMatch())
      return MCTargetAsmParser::Match_Success;
    break;
    }
```

**Pull request merged**